src/TLuaInterpreter.cpp

```diff
@@ -9202,98 +9281,25 @@ int TLuaInterpreter::createMapLabel(lua_State* L)
     bool noScaling = true;
 
     int args = lua_gettop(L);
-    if (!lua_isnumber(L, 1)) {
-        lua_pushstring(L, "createMapLabel: wrong argument type");
-        return lua_error(L);
-    }
-    int area = lua_tointeger(L, 1);
-
-    if (!lua_isstring(L, 2)) {
-        lua_pushstring(L, "createMapLabel: wrong argument type");
-        return lua_error(L);
-    }
-    QString text = lua_tostring(L, 2);
-
-    if (!lua_isnumber(L, 3)) {
-        lua_pushstring(L, "createMapLabel: wrong argument type");
-        return lua_error(L);
-    }
-    float posx = lua_tonumber(L, 3);
-
-    if (!lua_isnumber(L, 4)) {
-        lua_pushstring(L, "createMapLabel: wrong argument type");
-        return lua_error(L);
-    }
-    float posy = lua_tonumber(L, 4);
-
-    if (!lua_isnumber(L, 5)) {
-        lua_pushstring(L, "createMapLabel: wrong argument type");
-        return lua_error(L);
-    }
-    float posz = lua_tonumber(L, 5);
-
-    if (!lua_isnumber(L, 6)) {
-        lua_pushstring(L, "createMapLabel: wrong argument type");
-        return lua_error(L);
-    }
-    int fgr = lua_tointeger(L, 6);
-
-    if (!lua_isnumber(L, 7)) {
-        lua_pushstring(L, "createMapLabel: wrong argument type");
-        return lua_error(L);
-    }
-    int fgg = lua_tointeger(L, 7);
-
-    if (!lua_isnumber(L, 8)) {
-        lua_pushstring(L, "createMapLabel: wrong argument type");
-        return lua_error(L);
-    }
-    int fgb = lua_tointeger(L, 8);
-
-    if (!lua_isnumber(L, 9)) {
-        lua_pushstring(L, "createMapLabel: wrong argument type");
-        return lua_error(L);
-    }
-    int bgr = lua_tointeger(L, 9);
-
-    if (!lua_isnumber(L, 10)) {
-        lua_pushstring(L, "createMapLabel: wrong argument type");
-        return lua_error(L);
-    }
-    int bgg = lua_tointeger(L, 10);
-
-    if (!lua_isnumber(L, 11)) {
-        lua_pushstring(L, "createMapLabel: wrong argument type");
-        return lua_error(L);
-    }
-    int bgb = lua_tointeger(L, 11);
-
+    int area = getVerifiedInt(L, 1, "createMapLabel", "areaID");
+    QString text = getVerifiedString(L, 2, "createMapLabel", "text");
+    float posx = getVerifiedFloat(L, 3, "createMapLabel", "posX");
+    float posy = getVerifiedFloat(L, 4, "createMapLabel", "posY");
+    float posz = getVerifiedFloat(L, 5, "createMapLabel", "posZ");
+    int fgr = getVerifiedInt(L, 6, "createMapLabel", "fgRed");
+    int fgg = getVerifiedInt(L, 7, "createMapLabel", "fgGreen");
+    int fgb = getVerifiedInt(L, 8, "createMapLabel", "fgBlue");
+    int bgr = getVerifiedInt(L, 9, "createMapLabel", "bgRed");
+    int bgg = getVerifiedInt(L, 10, "createMapLabel", "bgGreen");
+    int bgb = getVerifiedInt(L, 11, "createMapLabel", "bgBlue");
```

Comment:
This way you don't have to type the function name all the time, which is less work, what do you think? 

```suggestion
    int area = getVerifiedInt(L, 1, __func__, "areaID");
    QString text = getVerifiedString(L, 2, __func__, "text");
    float posx = getVerifiedFloat(L, 3, __func__, "posX");
    float posy = getVerifiedFloat(L, 4, __func__, "posY");
    float posz = getVerifiedFloat(L, 5, __func__, "posZ");
    int fgr = getVerifiedInt(L, 6, __func__, "fgRed");
    int fgg = getVerifiedInt(L, 7, __func__, "fgGreen");
    int fgb = getVerifiedInt(L, 8, __func__, "fgBlue");
    int bgr = getVerifiedInt(L, 9, __func__, "bgRed");
    int bgg = getVerifiedInt(L, 10, __func__, "bgGreen");
    int bgb = getVerifiedInt(L, 11, __func__, "bgBlue");
```

`__func__` comes from C++11: https://en.cppreference.com/w/cpp/preprocessor/replace. You might also see `__FUNCTION__` be used in some places but that's the oldschool way before it became standard.

Reply:
The more you know..!